lib/common/mem.h

```diff
@@ -47,6 +47,23 @@ extern "C" {
 #define MEM_STATIC_ASSERT(c)   { enum { MEM_static_assert = 1/(int)(!!(c)) }; }
 MEM_STATIC void MEM_check(void) { MEM_STATIC_ASSERT((sizeof(size_t)==4) || (sizeof(size_t)==8)); }
 
+/* detects whether we are being compiled under msan */
+#if defined (__has_feature)
+#  if __has_feature(memory_sanitizer)
+#    define MEMORY_SANITIZER 1
+#  endif
+#endif
+
+#if defined (MEMORY_SANITIZER)
+#  include <sanitizer/msan_interface.h>
```

Comment:
Sadly, when compiling with `clang.par` and `clang++.par` on my devserver, it wasn't able to find this header. But, it did have the symbols, so declaring the functions we need here would work.

Repro:
```
sudo yum install fb-tp2-compilers
CC=clang.par CXX=clang.par ./fuzz.py build all --enable-fuzzer --enable-msan
```

Reply:
Oof. That sucks. Building on my desktop (clang-6.0 on Ubuntu 18.04.2) fails if I don't include the header (the symbols are not defined, and we disallow implicit definition). Not sure what an appropriate test is.

Reply:
Fixed by declaring the functions we need.